src/battle.c

```diff
@@ -1234,7 +1236,7 @@ static int apply_magicshield(int reduced_damage, fighter *df,
             /* jeder Schaden wird um effect% reduziert bis der Schild duration
              * Trefferpunkte aufgefangen hat */
             if (me->typ == SHIELD_REDUCE) {
-                int hp = reduced_damage * (me->effect / 100);
```

Comment:
Um das zu testen, ist es wohl am einfachsten, die ganze Logik für magische Reduktion in eine Funktion zu extrahieren, die ein meffect struct auf einen Schaden anwendet, und die Ausgaben zu testen. Das Refactoring dafür und den Test mache ich mal eben auch.